qiskit/ignis/verification/randomized_benchmarking/fitters.py

```diff
@@ -750,3 +763,189 @@ def plot_rb_data(self, pattern_index=0, ax=None,
+
+    @property
+    def cliff_lengths(self):
+        """Return clifford lengths."""
+        return self._cliff_lengths
+
+    @property
+    def fit(self):
+        """Return the purity fit parameters."""
+        return self.rbfit_pur[self._npurity].fit
+
+    @property
+    def fit_pur(self):
+        """Return the fit as a 3^n element list."""
+        return [self.rbfit_pur[d].fit for
+                d in range(self._npurity)]
+
+    @property
+    def rb_fit_fun(self):
+        """Return the function rb_fit_fun."""
+        return self.rbfit_pur[self._npurity] \
+            .rb_fit_fun
+
+    @property
+    def seeds(self):
+        """Return the number of loaded seeds as a
+        3^n element list."""
+        return [self.rbfit_pur[d].seeds for
+                d in range(self._npurity)]
+
+    @property
+    def results(self):
+        """Return all the results as a 3^n element list."""
+        return [self.rbfit_pur[d].results for
+                d in range(self._npurity)]
+
+    @property
+    def raw_data_pur(self):
+        """Return raw_data as a
+        3^n element list."""
+        return [self.rbfit_pur[d].raw_data for
+                d in range(self._npurity)]
+
+    @property
+    def raw_data(self):
+        """Return raw_data of the sum of 3^n
+        correlators"""
+        return self.rbfit_pur[self._npurity].raw_data
+
+    @property
+    def ydata(self):
+        """Return ydata (means and std devs)
+        of the sum of 3^n correlators"""
+        return self.rbfit_pur[self._npurity].ydata
+
+    def add_data(self, new_purity_result, rerun_fit=True):
+        """
+        Add a new result.
+        Args:
+            new_purity_result: list of purity rb results
+            rerun_fit: re-caculate the means and fit the result
+            Additional information:
+            Assumes that 'result' was executed is
+            the output of circuits generated by randomized_becnhmarking_seq
+            where is_purity = True.
+        """
+        for d in range(self._npurity):
+            self.rbfit_pur[d].add_data(new_purity_result[d], rerun_fit)
+
+        if rerun_fit:
+            self.calc_data()
+            self.calc_statistics()
+            self.fit_data()
+
+    def calc_data(self):
+        """
+        Retrieve probabilities of success from execution results.
+        Outputs results into an internal variables _raw_data.
+        Calculates the sum of the raw_data from the 3^n
+        correlators.
+        """
+        for d in range(self._npurity):
+            self.rbfit_pur[d].calc_data()
+
+        # Update the raw_data of the sum of correlators
+        self.rbfit_pur[self._npurity].raw_data = [
+            [[] for _ in range(len(self.seeds[0]))]
+            for _ in range(len(self._rb_pattern))]
+
+        for patt_ind in range(len(self._rb_pattern)):
+            for seed_ind in range(len(self.seeds[0])):
+                qubits = self._rb_pattern[patt_ind]
```

Comment:
Here's the main reason the separate fitters are not helping. This is not calculating the correlators. You are just summing the ground state populations from the different post rotation gates. The purity is the sum of all 4^n correlations *squared*. So you need further processing of the raw data to get these correlators and then you need to sum the square.

Reply:
You are right. Is it enough to replace these lines of code by the following code that sums the **squares**:
```
        for patt_ind in range(len(self._rb_pattern)):
            for seed_ind in range(len(self.seeds[0])):
                qubits = self._rb_pattern[patt_ind]
                new_raw_data = np.zeros(len(self._cliff_lengths[patt_ind]))
                for d in range(self._npurity):
                    new_raw_data += np.square(self.raw_data_pur
                                              [d][patt_ind][seed_ind])
                new_raw_data = new_raw_data / (2 ** len(qubits))
                self.rbfit_pur[self._npurity].raw_data[patt_ind][seed_ind] = \
                    new_raw_data
```

Reply:
No, because `self.raw_data_pur[d][patt_ind][seed_ind]` is not the correlator, it's just the ground state population

Reply:
So what exactly do you mean by correlator?

Reply:
For some Pauli's eg. P_i x P_j a correlator would be <P_i P_j>. There are 4^n correlators, and we have done 3^n experiments. You can get 2^n correlators per experiment, so there's some over-completeness.

Reply:
There's an example here https://github.ibm.com/IBMQuantum/qiskit-ignis-internal/blob/94432f9bfc7f616830028769f49f90cf9d609483/corr_rb/fitters.py#L919

Reply:
OK - I changed this